### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by tamminhdao**
I did some refactoring to address Cyrus's comment on the previous BasicAuth PR.
I now have each Action return a response, so I no longer have that terrible function in the Router with a million nested if statements.

This branch doesn't have any Auth code, this is the last point when the simultaneous test was still passing.

### CODE REVIEW

File: Sources/HttpServer/Actions/ActionProtocol.swift [hunk 1]
```diff
@@ -1,3 +1,3 @@
 public protocol HttpAction {
-    func execute(request: HttpRequest)
+    func execute(request: HttpRequest) -> HttpResponse
 }
```
Comment (cmvandrevala) [inline Sources/HttpServer/Actions/ActionProtocol.swift]: This is a minor detail - try to put a newline at the end of each file.

File: Sources/HttpServer/Actions/PostAction.swift [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
-    public func execute(request: HttpRequest) {
+    public func execute(request: HttpRequest) -> HttpResponse {
         let requestBody : [String: String] = request.returnBody()
         for item in requestBody {
             dataStorage.addValues(key: item.key, value: item.value)
         }
-        dataStorage.logValues()
+        return responseGenerator.generate200Response(method: HttpMethod.post, url: request.returnUrl())
     }
 }
```
Comment (cmvandrevala) [inline Sources/HttpServer/Actions/PostAction.swift]: Newline

File: Sources/HttpServer/Route/Route.swift [hunk 3]
```diff
@@ -3,9 +3,24 @@ public class Route {
...
+//    }
+//
+//    func put() {
+//        returns response
+//    }
+//
+//
+//    func options() {
+//
+//    }
```
Comment (cmvandrevala) [inline Sources/HttpServer/Route/Route.swift]: What are all of these comments doing here? Get rid of these!

File: Sources/HttpServer/Router/Router.swift [hunk 4]
```diff
@@ -8,6 +8,7 @@ public class Router {
     }
 
     public func route(request: HttpRequest) -> HttpResponse {
+
```
Comment (cmvandrevala) [inline Sources/HttpServer/Router/Router.swift]: Why the newline here?

File: Sources/HttpServer/Route/Route.swift (hunk 3 above)
Comment (tamminhdao) [reply Sources/HttpServer/Route/Route.swift]: Sorry. I was brainstorming and then forgot about these.

**Pull request closed without merging**